pkg/virt-operator/resource/apply/reconcile.go

```diff
@@ -136,6 +138,9 @@ func injectPlacementMetadata(componentConfig *v1.ComponentConfig, podSpec *corev
 	if _, ok := nodePlacement.NodeSelector[kubernetesOSLabel]; !ok {
 		nodePlacement.NodeSelector[kubernetesOSLabel] = kubernetesOSLinux
 	}
+	if _, ok := nodePlacement.NodeSelector[kubernetesArchLabel]; !ok {
```

Comment:
I can't tell from your change, is the runtime.GOARCH hard or soft affinity?

Reply:
This should be a hard affinity. `runtime.GOARCH` matches the platform of the `virt-operator` image and then the newly applied node selector makes sure the same set of `virt-handler`, `virt-api` and etc. could be run on the node with the corresponding arch.

Reply:
This PR is just to prevent the pods of `virt-operator` and pods of deployments/daemonsets created by `virt-operator` been scheduled on `arm64` nodes (the platform of my local build images is `linux/amd64`).

Reply:
That's my concern.  KubeVirt can create a control-plane daemonsets for both amd and arm.  But I guess that's not what you intended.

Reply:
From my understanding, the `arm64` support is still on its way. So currently, we can only choose to deploy KubeVirt on `amd64` or `arm64` (https://kubevirt.io/user-guide/operations/installation/#experimental-arm64-developer-builds). Please correct me if I am wrong.

Reply:
I think you can accomplish the same thing with labels?  
```
    infra:
      nodePlacement:
        affinity:
          nodeAffinity:
            requiredDuringSchedulingIgnoredDuringExecution:
              nodeSelectorTerms:
              - matchExpressions:
                - key:  kubernetes.io/arch
                  operator: In
                  values:
                  - amd64
```

Reply:
Yes, both `infra/nodePlacement` and `worloads/nodePlacement` need to be updated.